server/src/main/java/org/elasticsearch/transport/InboundPipeline.java

```diff
@@ -115,9 +115,13 @@ private void forwardFragment(TcpChannel channel, Object fragment) throws IOExcep
             InboundMessage aggregated = aggregator.finishAggregation();
             try {
                 statsTracker.markMessageReceived();
-                messageHandler.accept(channel, aggregated);
+                messageHandler.accept(channel, /* autocloses */ aggregated);
+                aggregated = null;
             } finally {
-                aggregated.decRef();
+                if (aggregated != null) {
+                    // TODO doesn't messageHandler auto-close always?
```

Comment:
looking at `messageHandler` (i.e. `TcpTransport#inboundMessage`) I think we close on all paths already, and catch everything anyway, so this seems redundant?

Reply:
Right ... let me fix that right up :)